addons/im_livechat/tests/test_user_livechat_username.py

```diff
@@ -81,3 +81,56 @@ def test_user_livechat_username_reactions(self):
                 "write_date": fields.Datetime.to_string(john.partner_id.write_date),
             },
         )
+
+
+class TestUserLivechatUsernameStore(HttpCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.operators = cls.env["res.users"].create(
+            [
+                {
+                    "name": "Operator 1",
+                    "login": "operator1",
+                    "livechat_username": "ELOPERADOR",
+                },
+                {
+                    "name": "Operator 2",
+                    "login": "operator2",
+                    "group_ids": [
+                        fields.Command.link(cls.env.ref("im_livechat.im_livechat_group_user").id),
+                        fields.Command.link(cls.env.ref("base.group_user").id),
+                    ],
+                },
+            ])
+        cls.livechat_channel = cls.env['im_livechat.channel'].create({
+            'name': 'The channel',
+            'user_ids': [(6, 0, cls.operators.ids)]
+        })
+
+    def setUp(self):
+        super().setUp()
+
+        def _compute_available_operator_ids(channel_self):
+            for record in channel_self:
+                record.available_operator_ids = record.user_ids
+
+        self.patch(type(self.env['im_livechat.channel']), '_compute_available_operator_ids', _compute_available_operator_ids)
```

Comment:
We should avoid hacking the method and create a proper setup for the expected result instead.